synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -292,6 +298,22 @@ async def _do_preview(self, url: str, user: str, ts: int) -> bytes:
 
             await self._precache_image_url(user, media_info, og)
 
+        elif oembed_url and _is_json(media_info.media_type):
+            # Handle an oEmbed response.
+            with open(media_info.filename, "r") as utf8_file:
+                oembed_body = utf8_file.read()
```

Comment:
It *should* be UTF-8 encoded JSON, I guess the headers could say they're something else, but that seems like it would be extremely rare? Maybe that's an invalid assumption though!

Reply:
I could also load this as bytes and push the `UTF-8` conversion into the oEmbed code, if you'd like? At least that way the exception should be caught more nicely.